src/search/SearchMessagesScreen.js

```diff
@@ -83,7 +84,7 @@ class SearchMessagesScreen extends PureComponent<Props, State> {
       messages,
       isFetching: this.lastIdSent !== this.lastIdReceived,
     });
-  };
+  }, 500);
```

Comment:
Perhaps shorter, like 200ms or 250ms? I imagine 500ms may feel pretty sluggish.

Easy to tweak later, though.

Reply:
Possibly. Average smartphone typing speed [seems to be ≤30 wpm](https://www.quora.com/What-is-the-average-smartphone-users-typing-speed), or [approximately](https://www.yorku.ca/mack/RN-TextEntrySpeed.html) 250ms/character, though; if that's accurate, a 250ms throttle sounds like it'd be functionally useless.

I *have* observed sluggishness, but it seems to have been due to `TextInput` itself delaying the messages. (Seen in simulation/emulation only; I haven't timed it on real hardware yet.)

Reply:
> Average smartphone typing speed [seems to be ≤30 wpm](https://www.quora.com/What-is-the-average-smartphone-users-typing-speed), or [approximately](https://www.yorku.ca/mack/RN-TextEntrySpeed.html) 250ms/character, though; if that's accurate, a 250ms throttle sounds like it'd be functionally useless.

Hmm. That speed does sound pretty plausible.

Still that doesn't mean it's the wrong value. There's always the possibility that whoever originally decided to stick a `throttle` call in here wasn't quite thinking it through. :wink: